### PR TITLE
Feature/all day events

### PR DESCRIPTION
**Pull request by TamarBerger**
Allows users to mark events as **all-day**. 
The Dayview display as well as the Event details display changes accordingly.

TODO: 
- Create a unique display for all-day events on the Week view HTML page.
- Make an automated fill for the _start time_, _end date_, and _end time_ fields, once the event is defined as all-day.

### CODE REVIEW

File: app/routers/event.py [hunk 1]
```diff
@@ -63,6 +63,12 @@ async def create_new_event(request: Request, session=Depends(get_db)):
     location_type = data['location_type']
     is_zoom = location_type == 'vc_url'
     location = data['location']
+    if data['event_type']:
+        all_day = data['event_type']
+        if all_day == "on":
+            all_day = True
+        else:
+            all_day = False
```
Comment (yammesicka) [inline app/routers/event.py:71]: `all_day = data['event_type'] and data['event_type'] == "on"`

File: app/templates/event/partials/edit_event_details_tab.html [hunk 2]
```diff
@@ -26,6 +26,16 @@
   <input type="text" id="invited" name="invited" placeholder="Invited emails, separated by commas">
 </div>
 
+<div class="form_row">
+    <select name="event_type" required>
+        <option value="" disabled selected>All Day</option>
+        <option value="on">Yes</option>
+        <option value="off">No</option>
+    </select>
```
Comment (yammesicka) [inline app/templates/event/partials/edit_event_details_tab.html:34]: Please add a label :)

**Review by yammesicka — COMMENTED** [review]
Great job! I've added few insights, please take a look :)

File: app/routers/event.py [hunk 3]
```diff
@@ -62,6 +62,11 @@ async def create_new_event(request: Request,
     owner_id = get_current_user(session).id
     availability = data.get('availability', 'True') == 'True'
     location = data['location']
+    # if data['event_type']:
+    #     all_day = data['event_type']
+    #     all_day = all_day == "on"
```
Comment (yammesicka) [inline app/routers/event.py:67]: If not necessary, please delete

File: app/routers/event.py [hunk 4]
```diff
@@ -87,16 +93,32 @@ async def create_new_event(request: Request,
...
+    'start': datetime,
+    'end': datetime,
+    'availability': bool,
+    'all_day': bool,
+    'content': (str, type(None)),
+    'location': (str, type(None)),
+    'category_id': (int, type(None))
+}
+
+
```
Comment (yammesicka) [inline app/routers/event.py:121]: Is this part needed?
If not - delete, if it is - please move to more proper location (if it's pydantic's annotations, there is a proper file for that)

File: app/routers/event.py [hunk 5]
```diff
@@ -87,16 +93,32 @@ async def create_new_event(request: Request,
 async def eventview(request: Request, event_id: int,
                     db: Session = Depends(get_db)) -> Response:
     event, comments, end_format = get_event_data(db, event_id)
+    start_format = START_FORMAT
+    if event.all_day is True:
```
Comment (yammesicka) [inline app/routers/event.py:97]: Just `event.all_day:`

**Review by yammesicka — COMMENTED** [review]
Great work! I've added few more insights :)

File: app/routers/dayview.py [hunk 6]
```diff
@@ -105,6 +105,20 @@ def _check_multiday_event(self) -> Tuple[bool]:
...
+        (event.end >= day and event.end < day_end) or
+        (event.start < day_end < event.end)
+        )
+
+
+def all_day_event_in_day(
+    event: Event, day: datetime, day_end: datetime
+) -> bool:
+    if not event.all_day:
+        return None
```
Comment (yammesicka) [inline app/routers/dayview.py:121]: Maybe `False`?

File: app/routers/dayview.py [hunk 7]
```diff
@@ -105,6 +105,20 @@ def _check_multiday_event(self) -> Tuple[bool]:
 
 
 def event_in_day(event: Event, day: datetime, day_end: datetime) -> bool:
+    if event.all_day:
+        return None
+    return (
+        (event.start >= day and event.start < day_end) or
+        (event.end >= day and event.end < day_end) or
+        (event.start < day_end < event.end)
+        )
```
Comment (yammesicka) [inline app/routers/dayview.py:114]: Isn't this couldn't be rephrased as `not all_day_event`?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/routers/dayview.py (hunk 7 above)
Comment (TamarBerger) [reply app/routers/dayview.py:114]: If I understand it correctly it checks whether the event occurs on a certain day or not- thus, can not be replaced...
I didn't write this function just added the conditioning:)

File: app/routers/dayview.py (hunk 6 above)
Comment (TamarBerger) [reply app/routers/dayview.py:121]: What do you mean?

Comment (yammesicka) [reply app/routers/dayview.py:121]: This function returns boolean and it's important to keep up the consistency.

**Review by yammesicka — CHANGES_REQUESTED** [review]
Excellent work!

File: tests/test_event.py [hunk 8]
```diff
@@ -218,6 +245,19 @@ def test_eventedit_post_correct(client, user):
...
 
+def test_eventedit_post_without_event_type(client, user):
+    """
+    Test create new event successfully,
+    When the event type is not defined by the user.
+    """
+    response = client.post(client.app.url_path_for('create_new_event'),
+                           data=CORRECT_EVENT_FORM_DATA)
+    assert response.ok
+    assert response.status_code == status.HTTP_302_FOUND
```
Comment (Gonzom) [inline tests/test_event.py:256]: if you are checking the status code then there is no need to also check .ok.

**Review by yammesicka — APPROVED** [review]

**Pull request merged**